lib/rubygems/ext/rake_builder.rb

```diff
@@ -26,8 +26,8 @@ def self.build(extension, dest_path, results, args=[], lib_dir=nil)
 
     rake ||= Gem.default_exec_format % 'rake'
 
-    cmd = "#{rake} RUBYARCHDIR=#{dest_path} RUBYLIBDIR=#{dest_path}" # ENV is frozen
-
+    cmd = "#{rake} RUBYARCHDIR=#{dest_path} RUBYLIBDIR=#{dest_path}".dup # ENV is frozen
+    cmd << " #{args.join " "}" unless args.empty?
```

Comment:
I don't think this will work if any of the args contain shell meta characters, such as spaces?

Reply:
I think you are probably right! Wanting to make the minimum possible change, I just copied this from the other builders. That seems to be how they do it.

Reply:
:sad_panda:

Reply:
Happy to do a followup PR that fixes this! But I think out of scope for this one since it would be a codebase-wide change.

Reply:
Yeah that’s fine, just sad to discover that we’re already doing the “wrong thing”

Reply:
@alyssais ^